distributions/R/scripts/new/header_parser.py

```diff
@@ -0,0 +1,171 @@
+    size_parameters = set()
+    for param_desc, param_doxy, i in zip(
+            fun['parameters'],
+            param_regex.finditer(fun['doxygen']),
+            range(len(fun['parameters']))):
+        param_mode = param_doxy.group('mode')
+        array_info = array_regex.search(param_doxy.group(0))
+        param_size = array_info.group('lname') if array_info else None
+        try:
+            param = parameters[param_desc['name']] = Parameter(param_desc, i, param_mode, param_size)
+        except ValueError as e:
+            raise ValueError(
+                'Cannot create wrapper for function '
+                f'{fun["name"]} (defined in {fun["filename"]}'
+            ) from e
+        if isinstance(param.size, str):
+            size_parameters.add(param.size)
+
+    list_parameters = sorted(parameters.values(), key=lambda x: x.ord)
+    out_params = [p for p in list_parameters if p.mode in ['out', 'in,out']]
+    if len(out_params) == 0:
+        warnings.warn(f'{fun["name"]} appears to have no output parameters (defined in {fun["filename"]}')
+        return ''  # no output parameters, no point in creating a wrapper
+    for param in size_parameters:
+        try:
+            if parameters[param].mode == 'in':
+                parameters[param].mode = 'size'
+        except KeyError as e:
+            raise ValueError(
+                f'{param} is not a parameter of '
+                f'{fun["name"]} (defined in {fun["filename"]}'
+            ) from e
+
+    for param in parameters.values():
+        if isinstance(param.size, str):
+            if param.mode in ['in', 'in,out']:
+                parameters[param.size].targets.append(param)
+
+    for param in size_parameters:
+        before_call.append(
+            f'\t{param} <- min({",".join([f"length({target.name})" for target in parameters[param].targets])})')
+
+    for param in list_parameters:
+        call_params.append(param.name)
+        if param.mode in ['in', 'in,out']:
+            parameter_list.append('\t' + param.name)
+            before_call.append(param.conversion)
+            if isinstance(param.size, int):
+                before_call.append(f'\tstopifnot(length({param.name}) >= {param.size})')
+        if param.mode in ['out', 'in,out']:
+            if param.mode == 'out':
+                before_call.append(
+                    f'\t{param.name} <- vector(mode = "'
+                    f'{"double" if param.type == "single" else param.type}'
+                    f'", length = {param.size if param.size else 1})'
+                )
+            after_call.append(f'\t{param.name} <- AUTO___RET__PARAMS${param.name}')
+
+    call_proper = call_proper + ', '.join(call_params) + ')'
+    if len(out_params) == 1:
+        after_call.append(f'\tAUTO__RETVAL <- {out_params[0].name}')
+    else:
+        after_call.append(f'''\tAUTO__RETVAL <- list({",".join([f'"{p.name}" = {p.name}' for p in out_params])})''')
+
+    return '\n'.join(signature
```

Comment:
Will this work for Windows ?
I'd suggest using os.sep or better specifying clearly what is the end-of-line character.
We may like to generate Windows-R script while working on Linux

Reply:
It works fine on Windows, at least with Rstudio. In fact, looking at other libraries it seems that just LF is something of a standard

Reply:
good !